src/umzug.ts

```diff
@@ -483,6 +480,23 @@ export class Umzug<Ctx extends object = object> extends emittery<UmzugEvents<Ctx
 		});
 	}
 
+	private async getContext(): Promise<Ctx> {
+		const isPromise = (_ctx: object | Function) => {
```

Comment:
Do we need this? Can't we just check if it's a function and `await context()` if it is?

Reply:
Unfortunately, Promise isn't a function and this comparison is required. We could use different mechanics to detect promise, such either `typeof _ctx.constructor === 'function';` or `Boolean(typeof value.then === 'function');`, but in my opinion, current implementation Is better way to do that. WDYT?

Reply:
I mostly mean that you can await a non-promise too. So we can keep lines 231-234 basically the same but put an `await` before each ternary statement option, then I don't think we'd even need this function.

Reply:
The context retrieving action is a significant one, and grouping it into the method `getContext` increases code readability. As for rest, if we would add `await` before each ternary, the Promise branch won't execute at all.

Reply:
That's true. I'm fine with keepinh getContext, encapsulating the logic in a function, as long as we get rid of this `toString` based promise checking. It's brittle (likely won't work with non-native promises like bluebird, etc.) and it doesn't seem necessary.

Reply:
I changed the Promise detection implementation. Please review.

Reply:
Any updates on it? :)

Reply:
I left a suggestion to explain what I mean. Please add a failing test if you think my suggestion doesn't cover everything.

Reply:
I simplified the method according to your suggestion. Thanks!